### ISSUE TITLE
how to list of glyph from figure

### ISSUE THREAD

**Issue by kkumawat25** [PatrikHlobil/Pandas-Bokeh#40]
Hi,
 How I can get a list of glyph so that I can do customization on each of the axes?
 I want to legends  to be outside the plot and below link provides a solution in bokeh, how to do this in pandas_bokeh ?

link : https://stackoverflow.com/questions/42646672/bokeh-pandas-legend-outside-plot

**Comment by PatrikHlobil** [comment]
Hi @kkumawat25 ,

The following works for me:

```python
import numpy as np
from bokeh.models import Legend
import pandas
import pandas_bokeh

np.random.seed(42)
df = pd.DataFrame({"Google": np.random.randn(1000)+0.2, 
                   "Apple": np.random.randn(1000)+0.17}, 
                   index=pd.date_range('1/1/2000', periods=1000))
df = df.cumsum()
df = df + 50
p = df.plot_bokeh(kind=“line”, legend=None, show_figure=False)
rs = p.renderers

names = ("Google", "Apple")
legend = Legend(items=[(name, [r]) for (name, r) in zip(names, rs)], 
       location=(0, 30))
p.add_layout(legend, "right")
pandas_bokeh.show(p)
```

Hope this helps you. Please give me feedback such that I can close this issue.

Best Patrik

**Comment by kkumawat25** [comment]
Hi Patrik,
 Thanks for a quick response. I am using bokeh 1.0.2.
I got below error in running above example.
AttributeError                            Traceback (most recent call last)
<ipython-input-6-10f4a6c4cb80> in <module>
     11 df = df.cumsum()
     12 df = df + 50
---> 13 df.plot_bokeh()
     14 # p = df.plot_bokeh(kind="line", legend=None, show_figure=False)
     15 # rs = p.renderers

C:\Anaconda3\envs\basecopy\lib\site-packages\pandas_bokeh\plot.py in __call__(self, *args, **kwargs)
   1735 
   1736     def __call__(self, *args, **kwargs):
-> 1737         return plot(self.df, *args, **kwargs)
   1738 
   1739     @property

C:\Anaconda3\envs\basecopy\lib\site-packages\pandas_bokeh\plot.py in df(self)
   1743             dataframe = self._parent
   1744         else:
-> 1745             dataframe = self._data
   1746 
   1747         # Convert PySpark Dataframe to Pandas Dataframe:

AttributeError: 'FramePlotMethods' object has no attribute '_data

**Comment by PatrikHlobil** [comment]
Hi @kkumawat25,

This issue us already reported in #39  and is relegated to pandas_bokeh==0.3. Try downgrading to pandas_bokeh==0.2 via:

    pip install pandas-bokeh==0.2

For the moment. I am currently in vacation and do not have my computer here, but I hope that I can fix the issue in the next week and make a  0.3.1 version that works just fine. Alternatively, try upgrading your pandas version to >=0.24. 

Best, Patrik
